### DISCUSSION TITLE
DynamicResource in code-behind does not take effect for SolidColorBrush.

### DISCUSSION THREAD

**Discussion by liwuqingxin** [AvaloniaUI/Avalonia#12847] [Q&A] [answered]
### Senario
I want to create a **color resource** and a **brush resource** from code. And I have checked the #4616.

- Firstly, I create the color resource whose key is '**ColorFromCodeBehind**' and add it to the styles's resources.
- Then, I create a SolidColorBrush object and bind it's ColorProperty to DynamicResourceExtension("**ColorFromCodeBehind**").
- At last, I add the styles object to the App's styles.

### Problem
The problem is the brush can not get the color dynamicly. 
> Note that the color resource does take effect. The brush that references to it in axaml performs well.

```csharp
public partial class App : Application
{
    public override void Initialize()
    {
        AvaloniaXamlLoader.Load(this);

        BuildResource();
    }

    private void BuildResource()
    {
        var styles = new Styles();

        var color = Color.Parse("Red");
        var dic = new ResourceDictionary();
        styles.Resources.MergedDictionaries.Add(dic);
        dic.Add("ColorFromCodeBehind", color);

        // Create the brush resource.
        var brush = new SolidColorBrush
        {
            [!SolidColorBrush.ColorProperty] = new DynamicResourceExtension("ColorFromCodeBehind")
        };
        dic.Add("BrushFromCodeBehind", brush);

        // Create a brush resource that binds to a Color resource defined in the axaml.
        var brush2 = new SolidColorBrush
        {
            [!SolidColorBrush.ColorProperty] = new DynamicResourceExtension("ColorFromXaml")
        };
        dic.Add("BrushFromCodeBehindButColorFromXaml", brush2);

        this.Styles.Add(styles);
    }
}

```

### Result

![image](https://github.com/AvaloniaUI/Avalonia/assets/20198528/566d989b-bd1a-449e-a021-75d6a8772b4d)

### Sample

[Avalonia.DynamicResource.Desktop.zip](https://github.com/AvaloniaUI/Avalonia/files/12568433/Avalonia.DynamicResource.Desktop.zip)

Any suggustions would be appreciated! 😄

**Comment by liwuqingxin** [comment]
🚀 Yes, I have overcome it. Just give an **anchor** to the DynamicResourceExtension. The **Appliction.Current** is an IResourceHost.

```csharp
var field = typeof(DynamicResourceExtension).GetField("_anchor", BindingFlags.NonPublic | BindingFlags.Instance);
field!.SetValue(dynamicResource, Application.Current);
```

**Comment by timunie** [comment, marked as answer]
actually, you can simply do:

```cs
var brush = new SolidColorBrush
{
    [!SolidColorBrush.ColorProperty] = dic.GetResourceObservable("TestColor").ToBinding()
};

// or

brush.Bind(SolidColorBrush.ColorProperty,  dic.GetResourceObservable("TestColor"));
```
